// hello_world.proto
syntax = "proto3";

package helloworld;

// Définir le service
service HelloWorldService {
  // RPC qui prend en entrée la langue et renvoie le message
  rpc SayHello (HelloRequest) returns (HelloResponse);
  rpc SayHelloManyTimes (HelloStreamRequest) returns (stream HelloStreamResponse);
}

// Définir les messages
message HelloRequest {
  string language = 1;
}

message HelloResponse {
  string message = 1;
}
message HelloStreamRequest {
  string language = 1;
  int32  count = 2;
  int32  intervalMS = 3; // in milliseconds

}

message HelloStreamResponse {
  string message = 1;
  string time = 2;
}

